prov/shm/src/smr_ep.c

```diff
@@ -319,6 +319,86 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
 }
 
+int smr_mmap_copy_format(struct smr_ep *ep, struct smr_region *peer_smr,
+		    struct smr_cmd *cmd, fi_addr_t peer_id,
+		    const struct iovec *iov, size_t count, size_t total_len,
+		    uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
+		    void *context, struct smr_resp *resp,
+		    struct smr_tx_entry *pend)
+{
+	char shm_name[NAME_MAX];
+	void *mapped_ptr;
+	int fd, ret, num;
+	uint64_t msg_id;
+	struct smr_ep_name *map_name;
+
+	msg_id = ep->msg_id++;
+	num = snprintf(shm_name, NAME_MAX - 1, "%s_%s_%ld", ep->name,
+		       (char *)peer_smr + peer_smr->name_offset, msg_id);
```

Comment:
Why is the name "name1_name2_id"? Shouldn't just having the initiator peer name be enough?

Reply:
Yeah. Since we are using per endpoint msg_id instead of per peer msg_id, having the initiator peer name will suffice.